src/handler.rs

```diff
@@ -172,11 +172,64 @@ where
     }
 }
 
-impl<A, M, I: 'static, E: 'static> MessageResponse<A, M> for ResponseFuture<Result<I, E>>
+/// MessageResponse trait impl to enbale the use of any I: 'static with Actor Handlers
+/// Usage with Result<I,E>:
+/// ```
+/// # pub struct MyActorAsync {}
+/// # impl Actor for MyActorAsync { type Context = actix::Context<Self>; }
+/// # use actix::prelude::*;
+/// # use core::pin::Pin;
+/// 
+/// pub struct MyQuestion{}
+/// impl Message for MyQuestion {
+///     type Result = Result<u8,u8>;
+/// }
+/// impl Handler<MyQuestion> for MyActorAsync {
+///     type Result = Pin<Box<dyn std::future::Future<Output = Result<u8,u8> >>>;
```

Comment:
Box and Pin are less likely to change aren't they?
Besides ResponseFuture is a bit redundant at this point, all it does it attempts to hide the fact that it's really Pin<Box<>>, and that doesn't help understanding the code.

Reply:
The types `Box` and `Pin` are themselves less likely to change, yes, however the the implementation is not made on them. The implementation is made over ResponseFuture, which right now is just an type alias but we could change it to a `struct`of it's own at some point and having a alias for it would result in less coding rewriting if that happens. The alias is meant to make code more easy to read, it's hard to evaluated rather it does that well, but it does indeed makes it shorter.